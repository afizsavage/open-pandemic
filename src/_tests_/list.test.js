import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';

import ListComponent from '../components/Countries/list-component';
import store from '../redux/configure-store';

describe('Test unordered list compoent', () => {
  it('Renders as expected', () => {
    const testObject = { id: 'twenty', today_confirmed: 234, image: '../images/afghanistan.svg' };
    const testProps = { array: [testObject], id: 'first-list' };

    const component = renderer.create(
      <Provider store={store}>
        <Router><ListComponent array={testProps} /></Router>
      </Provider>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
