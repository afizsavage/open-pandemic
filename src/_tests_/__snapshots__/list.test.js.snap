// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test unordered list compoent Renders as expected 1`] = `
<ul
  className="flex column"
  id="first-list"
>
  <li
    className="relative b-fourth"
  >
    <svg
      className="absolute white-text"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
        fillRule="evenodd"
      />
    </svg>
    <div
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="top flex center-x"
      >
        <img
          alt="Country Flag"
          className="country-image"
          src="../images/afghanistan.svg"
        />
      </div>
      <div
        className="bottom text-right white-text flex column"
      >
        <span
          className="count-name"
        />
        <span>
          234
        </span>
      </div>
    </div>
  </li>
</ul>
`;
